fearless/Modules/Staking/ControllerAccount/ControllerAccountInteractor.swift

```diff
@@ -0,0 +1,137 @@
+
+    private var stashItemProvider: StreamableProvider<StashItem>?
+    private var accountInfoProvider: AnyDataProvider<DecodedAccountInfo>?
+    private var ledgerProvider: AnyDataProvider<DecodedLedgerInfo>?
+    private var extrinsicService: ExtrinsicServiceProtocol?
+
+    init(
+        singleValueProviderFactory: SingleValueProviderFactoryProtocol,
+        substrateProviderFactory: SubstrateDataProviderFactoryProtocol,
+        runtimeService: RuntimeCodingServiceProtocol,
+        selectedAccountAddress: AccountAddress,
+        accountRepository: AnyDataProviderRepository<AccountItem>,
+        operationManager: OperationManagerProtocol,
+        feeProxy: ExtrinsicFeeProxyProtocol,
+        extrinsicServiceFactory: ExtrinsicServiceFactoryProtocol
+    ) {
+        self.singleValueProviderFactory = singleValueProviderFactory
+        self.substrateProviderFactory = substrateProviderFactory
+        self.runtimeService = runtimeService
+        self.selectedAccountAddress = selectedAccountAddress
+        self.accountRepository = accountRepository
+        self.operationManager = operationManager
+        self.feeProxy = feeProxy
+        self.extrinsicServiceFactory = extrinsicServiceFactory
+    }
+
+    private func handleStash(accountItem: AccountItem) {
+        extrinsicService = extrinsicServiceFactory.createService(accountItem: accountItem)
+        do {
+            let setController = try callFactory.setController(accountItem.address)
+            let identifier = setController.callName + accountItem.identifier
+
+            feeProxy.estimateFee(using: extrinsicService!, reuseIdentifier: identifier) { builder in
+                try builder.adding(call: setController)
+            }
+        } catch {
+            presenter.didReceiveFee(result: .failure(error))
+        }
+    }
+}
+
+extension ControllerAccountInteractor: ControllerAccountInteractorInputProtocol {
+    func setup() {
+        stashItemProvider = subscribeToStashItemProvider(for: selectedAccountAddress)
+
+        fetchAllAccounts(
+            from: accountRepository,
+            operationManager: operationManager
+        ) { [weak self] result in
+            self?.presenter.didReceiveAccounts(result: result)
+        }
+        feeProxy.delegate = self
+    }
+
+    func estimateFee(for controllerAddress: AccountAddress) {
+        do {
+            let setController = try callFactory.setController(controllerAddress)
+            let identifier = setController.callName + controllerAddress
+
+            feeProxy.estimateFee(using: extrinsicService!, reuseIdentifier: identifier) { builder in
+                try builder.adding(call: setController)
+            }
+        } catch {
+            presenter.didReceiveFee(result: .failure(error))
+        }
+    }
+
+    func fetchLedger(controllerAddress: AccountAddress) {
+        clear(dataProvider: &ledgerProvider)
+        ledgerProvider = subscribeToLedgerInfoProvider(
```

Comment:
It won't work since we subscribe only to current stashItem in blockchain. So, I suggest to use ```StorageRequestFactory``` to manually fetch the ledger.